File: app/src/main/java/cn/iome/faceme/utility/UIUtil.java
package cn.iome.faceme.utility;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;

import cn.iome.faceme.function.Consumer;

/**
 * Created by haoping on 17/5/3.
 * TODO
 */
public class UIUtil {

    public static void showDialog(Context context, String title, String message, String positive, String negative, boolean canceledOnTouchOutside, final Consumer<DialogInterface> callback) {
        final AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(title);
        builder.setMessage(message);
        builder.setPositiveButton(positive, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                callback.accept(dialog);
            }
        });
        builder.setNegativeButton(negative, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                callback.accept(dialog);
            }
        });
        AlertDialog alertDialog = builder.create();
        alertDialog.setCanceledOnTouchOutside(canceledOnTouchOutside);
        alertDialog.show();
    }

}

File: fakotlin/src/main/java/cn/iome/fakotlin/utility/UIUtil.kt
package cn.iome.fakotlin.utility

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface

import cn.iome.fakotlin.function.Consumer

/**
 * Created by haoping on 17/5/3.
 * TODO
 */
object UIUtil {

    fun showDialog(context: Context, title: String, message: String, positive: String, negative: String, canceledOnTouchOutside: Boolean, callback: Consumer<DialogInterface>) {
        val builder = AlertDialog.Builder(context)
        builder.setTitle(title)
        builder.setMessage(message)
        builder.setPositiveButton(positive) { dialog, which -> callback.accept(dialog) }
        builder.setNegativeButton(negative) { dialog, which -> callback.accept(dialog) }
        val alertDialog = builder.create()
        alertDialog.setCanceledOnTouchOutside(canceledOnTouchOutside)
        alertDialog.show()
    }

}
